### PR TITLE
Support management sdk call rest apis via AAD Token

### PR DESCRIPTION
**Pull request by terencefan**
### Summary of the changes (Less than 80 chars)
 - Detail 1
 - Detail 2

Fixes #bugnumber (in this specific format)

### CODE REVIEW

File: AzureSignalR.sln [hunk 1]
```diff
@@ -71,7 +71,7 @@ Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "Microsoft.Azure.SignalR.Int
 EndProject
 Project("{2150E333-8FDC-42A3-9474-1A3956D46DE8}") = "emulator", "emulator", "{ADD61A03-340D-426C-BA51-DBBC096434ED}"
 EndProject
-Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "Microsoft.Azure.SignalR.Emulator", "src\Microsoft.Azure.SignalR.Emulator\Microsoft.Azure.SignalR.Emulator.csproj", "{ABAAAEA8-40B3-4178-A7FD-09779A5DC757}"
+Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "Microsoft.Azure.SignalR.Emulator", "src\Microsoft.Azure.SignalR.Emulator\Microsoft.Azure.SignalR.Emulator.csproj", "{ABAAAEA8-40B3-4178-A7FD-09779A5DC757}"
```
Comment (wanlwanl) [inline AzureSignalR.sln:74]: Why change it?

File: test/Microsoft.Azure.SignalR.Management.Tests/AadAuthFacts.cs [hunk 2]
```diff
@@ -0,0 +1,24 @@
+﻿using System.Threading.Tasks;
+using Microsoft.Extensions.Logging;
+using Xunit;
+
+namespace Microsoft.Azure.SignalR.Management.Tests
+{
+    public class AadAuthFacts
+    {
+        [Fact(Skip = "")]
```
Comment (wanlwanl) [inline test/Microsoft.Azure.SignalR.Management.Tests/AadAuthFacts.cs:9]: How about update `appsettings.Test.json` and put it to e2e test project? Or use the mock service?

File: AzureSignalR.sln (hunk 1 above)
Comment (terencefan) [reply AzureSignalR.sln:74]: I'll revert it later

File: test/Microsoft.Azure.SignalR.Management.Tests/AadAuthFacts.cs (hunk 2 above)
Comment (terencefan) [reply test/Microsoft.Azure.SignalR.Management.Tests/AadAuthFacts.cs:9]: deleted and will be moved to E2E test.

File: src/Microsoft.Azure.SignalR.Management/ServiceManagerOptions.cs [hunk 3]
```diff
@@ -39,7 +59,10 @@ public class ServiceManagerOptions
 
         internal void ValidateOptions()
         {
-            ValidateConnectionString();
+            if (ServiceEndpoint == null)
+            {
+                ValidateConnectionString();
+            }
```
Comment (terencefan) [inline src/Microsoft.Azure.SignalR.Management/ServiceManagerOptions.cs:65]: TODO

File: src/Microsoft.Azure.SignalR.Common/Endpoints/ServiceEndpoint.cs [hunk 4]
```diff
@@ -39,13 +39,14 @@ public class ServiceEndpoint
...
         {
-            // Endpoint = string.Format("https://{0}.service.signalr.net", resourceName);
             Endpoint = endpoint;
             AccessKey = new AadAccessKey(authOptions);
 
             Version = "1.0";
             Port = port;
             Name = "";
+
+            EndpointType = EndpointType.Primary;
```
Comment (vicancy) [inline src/Microsoft.Azure.SignalR.Common/Endpoints/ServiceEndpoint.cs:49]: do you want the endpoint type as a ctor parameter?

**Review by vicancy — APPROVED** [review]

File: src/Microsoft.Azure.SignalR.Management/ServiceManagerOptions.cs [hunk 5]
```diff
@@ -22,6 +21,27 @@ public class ServiceManagerOptions
         /// </summary>
         public string ConnectionString { get; set; } = null;
```
Comment (Y-Sindo) [inline src/Microsoft.Azure.SignalR.Management/ServiceManagerOptions.cs:22]: If a user sets a service endpoint and then sets a connection string, then the connection string does not take effect, which is not excepted. I think setting service endpoint each time setting connection string is better?

**Pull request merged**